### PR TITLE
Enhancement/86epe342b/cs drop down cloze questionnaire blocks

### PR DESCRIPTION
**Pull request by acostales08**

### CODE REVIEW

File: apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DDCloze/DDCloze.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { Paper, Grid } from '@mui/material';
+import { TextField, MenuItem } from '@mui/material';
+import NearMeIcon from '@mui/icons-material/NearMe';
+import { DDClozeProps, QuestionaireProps, AnswerProps, OptionType } from "@/core/types/ssrData";
+
+
+export const DDCloze: React.FC<DDClozeProps> = ({ questionaire, answer, selectedValues, handleSelectChange }) => {
+        const renderDropdown = (options: OptionType[], selectFieldKey: string) => {
```
Comment (JMSevilla) [inline apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DDCloze/DDCloze.tsx:9]: Transfer this in a separate file in core > constant

**Review by JMSevilla — CHANGES_REQUESTED** [review]

File: apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DDCloze/DDCloze.tsx (hunk 1 above)
Comment (acostales08) [reply apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DDCloze/DDCloze.tsx]: dropdown reimplement ,renderDropdown function deleted

File: apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDQuestionaire.tsx
Comment (DropitupQwerty) [inline apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDQuestionaire.tsx:1]: Delete this file. This file is already exist in DND directory

File: apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DDCloze/DDCloze.tsx [hunk 2]
```diff
@@ -85,17 +55,31 @@ export const DDCQuestion: React.FC<SsrData> = ({ questionaire, answer }) => {
                             </div>
                             <Paper elevation={3} className="p-5 overflow-auto flex flex-col gap-5">
                                 <div className="w-full h-fit text-sm p-1">
-                                    {answerItem.DDCAnswer.split(/\[\[selectFieldKey:(\w+)\]\]/g).map((part, index) => {
+                                    {answerItem.DDCAnswer.split(/\[\[selectFieldKey:(\w+)\]\]/g).map((part: string, index: number) => {
                                         if (index % 2 === 0) {
```
Comment (DropitupQwerty) [inline apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DDCloze/DDCloze.tsx:59]: Avoid using if condition inside mapping

**Review by DropitupQwerty — APPROVED** [review]

**Review by JMSevilla — APPROVED** [review]

**Review by JezChristian — APPROVED** [review]

**Pull request merged**